sql/core/src/main/scala/org/apache/spark/sql/execution/QueryExecution.scala

```diff
@@ -54,6 +54,8 @@ class QueryExecution(
     val logical: LogicalPlan,
     val tracker: QueryPlanningTracker = new QueryPlanningTracker) extends Logging {
 
+  val id: UUID = UUID.randomUUID()
```

Comment:
Should we maybe follow who we do in `Dataset.id` or `SQLExecution.nextExecutionId`?

Reply:
I guess you meant using monotonically incremental long value; yes that would also work. Will update.